backend/app/core/rag/graph_builder.py

```
@@ -1,132 +1,156 @@
+    path.parent.mkdir(parents=True, exist_ok=True)
+    with open(path, "wb") as file:
+        pickle.dump(graph, file)
+
+
+def load_graph_from_disk(folder_path: str) -> Optional[nx.Graph]:
+    path = get_graph_cache_path(folder_path)
+    if not path.exists():
+        return None
+
+    with open(path, "rb") as file:
+        return pickle.load(file)
```

Ruff (0.15.12)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)